### PR TITLE
[merge script] Check if the CI that was run is outdated.

### LINKED ISSUE #7718: We should avoid merging PRs with outdated CI
Recently, we have seen quite a few failures due to PRs with stale successful CI being merged. We need to avoid this.

Ideally, we would have a bot comparing the md5 sum of the CI files and adding a `needs:rebase` label to those outdated.

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** infrastructure.

Fixes / closes #7718

We warn the merger that the CI has changed between the head of the branch the PR was based on and the current head of the branch. We give them the possibility to view the diff, then to decide whether to ignore the warning or request a rebase.

In the present state of this PR, we are diffing on `.travis.yml .gitlab-ci.yml appveyor.yml` only. If considered useful we could add `dev/ci/ ':!dev/ci/user-overlays/'` (that is everything in `dev/ci` except `dev/ci/user-overlays/` but this will make the diffs significantly larger and the present state already catches the kind of problem discussed in #7718).

I'll keep coq/bot#6 open to remind that this is only a partial solution and ideally @coqbot could help to anticipate when a PR should be rebased because it has an outdated CI.

### CODE REVIEW

File: dev/tools/merge-pr.sh [hunk 1]
```diff
@@ -140,6 +140,24 @@ if [ "$LOCAL_BRANCH_COMMIT" != "$UPSTREAM_COMMIT" ]; then
     fi
 fi
 
+# Sanity check: PR has an outdated version of CI
+
+BASE_COMMIT=$(echo "$PRDATA" | jq -r '.base.sha')
+CI_FILES=".travis.yml .gitlab-ci.yml appveyor.yml"
```
Comment (SkySkimmer) [inline dev/tools/merge-pr.sh]: Use an array instead of messing with space splitting.

**Comment by ejgallego** [comment]
Great!

**Comment by Zimmi48** [comment]
@SkySkimmer 

> Use an array instead of messing with space splitting.

Like this?

**Review by SkySkimmer — APPROVED** [review]

**Review by ejgallego — APPROVED** [review]
Very good.

**Review by maximedenes — APPROVED** [review]
I haven't tested it, but I like the idea. Thanks!

**Pull request merged**